File: Source/AlphaTab/Audio/Synth/Synthesis/Channel.cs
namespace AlphaTab.Audio.Synth.Synthesis
{
    internal class Channel
    {
        public ushort PresetIndex { get; set; }
        public ushort Bank { get; set; }
        public ushort PitchWheel { get; set; }
        public ushort MidiPan { get; set; }
        public ushort MidiVolume { get; set; }
        public ushort MidiExpression { get; set; }
        public ushort MidiRpn { get; set; }
        public ushort MidiData { get; set; }
        public float PanOffset { get; set; }
        public float GainDb { get; set; }
        public float PitchRange { get; set; }
        public float Tuning { get; set; }

        public float MixVolume { get; set; }
        public bool Mute { get; set; }
        public bool Solo { get; set; }
    }
}

File: packages/alphatab/src/synth/synthesis/Channel.ts
// The SoundFont loading and Audio Synthesis is based on TinySoundFont, licensed under MIT,
// developed by Bernhard Schelling (https://github.com/schellingb/TinySoundFont)
// TypeScript port for alphaTab: (C) 2020 by Daniel Kuschny
// Licensed under: MPL-2.0

export class Channel {
    public presetIndex: number = 0;
    public bank: number = 0;
    public pitchWheel: number = 0;
    public perNotePitchWheel: Map<number, number> = new Map<number, number>();
    public midiPan: number = 0;
    public midiVolume: number = 0;
    public midiExpression: number = 0;
    public midiRpn: number = 0;
    public midiData: number = 0;
    public panOffset: number = 0;
    public gainDb: number = 0;
    public pitchRange: number = 0;
    public tuning: number = 0;
    public mixVolume: number = 0;
    public mute: boolean = false;
    public solo: boolean = false;
}
